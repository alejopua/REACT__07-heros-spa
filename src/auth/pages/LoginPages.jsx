import { useNavigate } from 'react-router-dom';

export const LoginPages = () => {
  const navigate = useNavigate()

  const onLogout = () => {
      navigate('/', { replace: true })
  }

  return (
    <div className="container mt-5">
      <h1>Login</h1>
      <hr />

      <button 
        className="btn btn-primary"
        onClick={onLogout}
      >
        Login
      </button>
    </div>
  )
}
