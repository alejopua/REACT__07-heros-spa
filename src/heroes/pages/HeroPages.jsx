import { Navigate, useParams } from "react-router-dom"
import { getHeroeById } from "../helpers"

export const HeroPages = () => {
  const { id } = useParams()
  const hero = getHeroeById(id)

  if (!hero) return <Navigate to="/marvel" />

  return (
    <h1>HeroPages</h1>
  )
}
