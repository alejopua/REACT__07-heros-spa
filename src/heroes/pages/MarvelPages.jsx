import { HeroList } from "../components/HeroList"

export const MarvelPages = () => {
  return (
    <>
      <h1>MarvelPages</h1>
      <hr />
      <HeroList publisher="Marvel Comics" />
    </>
  )
}
