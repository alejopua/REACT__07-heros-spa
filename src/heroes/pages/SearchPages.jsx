import { useLocation, useNavigate } from 'react-router-dom'
import queryString from 'query-string'
import { useForm } from "../../hooks/useForm"
import { HeroCard } from "../components/HeroCard"
import { getHeroesByName } from '../helpers'

export const SearchPages = () => {
  const navigate = useNavigate()
  const location = useLocation()

  const { q = ''} = queryString.parse( location.search )
  const heroes = getHeroesByName( q )

  const { searchText, onInputChange} = useForm({ searchText: q.length > 0 ? q : ''})

  const onSearchSubmit = (e) => {
    e.preventDefault()
    navigate(`?q=${ searchText.toLocaleLowerCase().trim() }`)
  }

  return (
    <>
      <h1>Search</h1>
      <hr />

      <div className="row">
        <div className="col-5">
          <form onSubmit={ onSearchSubmit }>
            <input 
              type="text" 
              name="searchText" 
              id="search" 
              className="form-control" 
              placeholder="Find your hero..."
              autoComplete="off"
              value={ searchText }
              onChange={ onInputChange }
            />
            <button 
              type="submit" 
              className="btn mt-2 btn-block btn-outline-primary"
            >Search</button>
          </form>
        </div>

        <div className="col-7">
          <div className="animate__animated animate__fadeIn">

            {
              ( q === '' ) 
                ? <div className="alert alert-info">Search a hero</div>
                : ( heroes.length === 0) && <div className="alert alert-danger">Hero <b>{ q }</b> not encountered</div>
            }

            {
              heroes.map( hero => (
                <HeroCard key={ hero.id } {...hero} />
              ))
            }

          </div>
        </div>

      </div>
    </>
  )
}
