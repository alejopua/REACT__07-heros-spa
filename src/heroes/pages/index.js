export * from './DCPages'
export * from './MarvelPages'