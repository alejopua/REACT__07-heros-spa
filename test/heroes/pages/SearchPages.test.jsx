import { render, screen } from "@testing-library/react";
import { MemoryRouter } from 'react-router-dom'
import { SearchPages } from "../../../src/heroes/pages/SearchPages";

describe('Name of the group', () => {
  
  test('Should show Batman and the input with the querystring value.', () => {
    
    render(
      <MemoryRouter initialEntries={['/search?q=batman']}>
        <SearchPages />
      </MemoryRouter>
    )

    const input = screen.getByRole('textbox')
    expect( input.value ).toBe('batman')

    const imgHero = screen.getByRole('img')
    expect( imgHero.src ).toContain('/heroes/dc-batman.jpg')
  });

});