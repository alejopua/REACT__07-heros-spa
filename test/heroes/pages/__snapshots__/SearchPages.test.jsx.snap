// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Name of the group Should show correctly with the default values. 1`] = `
<div>
  <h1>
    Search
  </h1>
  <hr />
  <div
    class="row"
  >
    <div
      class="col-5"
    >
      <form>
        <input
          autocomplete="off"
          class="form-control"
          id="search"
          name="searchText"
          placeholder="Find your hero..."
          type="text"
          value=""
        />
        <button
          class="btn mt-2 btn-block btn-outline-primary"
          type="submit"
        >
          Search
        </button>
      </form>
    </div>
    <div
      class="col-7"
    >
      <div
        class="animate__animated animate__fadeIn"
      >
        <div
          class="alert alert-info"
        >
          Search a hero
        </div>
      </div>
    </div>
  </div>
</div>
`;
